File: packages/command-client/src/tauri/flow.ts
import { call } from "./invoke";
import type { FlowDefinition, FlowRun, FlowRunInput } from "../types/flow";
export const listFlows = (workspaceId: string) =>
  call<FlowDefinition[]>("flow_list", { workspaceId });
export const getFlow = (workspaceId: string, flowId: string) =>
  call<FlowDefinition>("flow_get", { workspaceId, flowId });
export const saveFlow = (input: FlowDefinition) =>
  call<FlowDefinition>("flow_save", { input });
export const deleteFlow = (workspaceId: string, flowId: string) =>
  call<void>("flow_delete", { workspaceId, flowId });
export const runFlow = (input: FlowRunInput) =>
  call<FlowRun>("flow_run", { input });
export const listFlowRuns = (workspaceId: string, flowId: string) =>
  call<FlowRun[]>("flow_runs_list", { workspaceId, flowId });
export const getFlowRun = (workspaceId: string, runId: string) =>
  call<FlowRun>("flow_run_get", { workspaceId, runId });
export const cancelFlowRun = (workspaceId: string, runId: string) =>
  call<FlowRun>("flow_run_cancel", { workspaceId, runId });

File: crates/unfour-app/src/commands/flow.rs
use crate::AppState;
use tauri::State;
use unfour_core::{models::*, AppResult};

#[tauri::command]
pub async fn flow_list(
    workspace_id: String,
    state: State<'_, AppState>,
) -> AppResult<Vec<FlowDefinition>> {
    state.command_bus.list_flows(workspace_id).await
}

#[tauri::command]
pub async fn flow_get(
    workspace_id: String,
    flow_id: String,
    state: State<'_, AppState>,
) -> AppResult<FlowDefinition> {
    state.command_bus.get_flow(workspace_id, flow_id).await
}

#[tauri::command]
pub async fn flow_delete(
    workspace_id: String,
    flow_id: String,
    state: State<'_, AppState>,
) -> AppResult<()> {
    state.command_bus.delete_flow(workspace_id, flow_id).await
}

#[tauri::command]
pub async fn flow_runs_list(
    workspace_id: String,
    flow_id: String,
    state: State<'_, AppState>,
) -> AppResult<Vec<FlowRunSummary>> {
    state
        .command_bus
        .list_flow_runs(workspace_id, flow_id)
        .await
}

#[tauri::command]
pub async fn flow_run_get(
    workspace_id: String,
    run_id: String,
    state: State<'_, AppState>,
) -> AppResult<FlowRun> {
    state.command_bus.get_flow_run(workspace_id, run_id).await
}

#[tauri::command]
pub async fn flow_run_cancel(
    workspace_id: String,
    run_id: String,
    state: State<'_, AppState>,
) -> AppResult<FlowRun> {
    state
        .command_bus
        .cancel_flow_run(workspace_id, run_id)
        .await
}

#[tauri::command]
pub async fn flow_save(
    input: FlowDefinition,
    state: State<'_, AppState>,
) -> AppResult<FlowDefinition> {
    state.command_bus.save_flow(input).await
}

#[tauri::command]
pub async fn flow_run(input: FlowRunInput, state: State<'_, AppState>) -> AppResult<FlowRun> {
    state.command_bus.run_flow(input).await
}
